syscall/syscall.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build darwin
+// +build darwin
+
+package syscall
+
+import "unsafe"
+
+func SyscallX(fn, a1, a2, a3 uintptr) (r1, r2, err uintptr) {
```

Comment:
Like Windows, would `SyscallN` taking arbitrary number of arguments more useful?

https://go-review.googlesource.com/c/go/+/336550

Reply:
It probably would be. I'll try to write it but I was having a strange bug occur where IsFramebuffer was returning false when I tried to consolidate the SyscallX into Syscall6X. Perhaps, a current solution would be to just switch over the number of arguments and call either of the three functions and worry about writing one assembly function later?

Reply:
> Perhaps, a current solution would be to just switch over the number of arguments and call either of the three functions and worry about writing one assembly function later?

Sounds good!